jaxws-ri/tools/wscompile/src/test/java/com/sun/tools/ws/test/processor/modeler/annotation/NillableArrayTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Distribution License v. 1.0, which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+package com.sun.tools.ws.test.processor.modeler.annotation;
+
+import com.sun.tools.ws.wscompile.WsgenTool;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import javax.xml.parsers.DocumentBuilderFactory;
+import junit.framework.Assert;
+import junit.framework.TestCase;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+public class NillableArrayTest extends TestCase {
+    public void testNillableArray() {
+        File destDir;
+        List<String> options;
+
+        destDir = new File(System.getProperty("java.io.tmpdir"), NillableArrayTest.class.getSimpleName());
+        destDir.mkdirs();
+        options = new ArrayList<>();
+        options.add("-d");
+        options.add(destDir.getAbsolutePath());
+        options.add("-cp");
+        options.add(System.getProperty("java.class.path") + File.pathSeparator + System.getProperty("jdk.module.path"));
+        options.add("com.sun.tools.ws.test.processor.modeler.annotation.NillableTest");
+        options.add("-wsdl");
+
+        Properties props = System.getProperties();
+        props.setProperty("com.sun.xml.ws.jaxb.allowNonNillableArray","true");
+
+        WsgenTool wsgen = new WsgenTool(System.out);
+        wsgen.run(options.toArray(new String[options.size()]));
+
+        //resetting system property com.sun.xml.ws.jaxb.allowNonNillableArray
+        //to null as it was before running the testcase 
+        props.setProperty("com.sun.xml.ws.jaxb.allowNonNillableArray","");

```

Comment:
Will this line be executed if the call to `wsgen.run(...)` throws unchecked exception?

A call to `System.clearProperty(...)` in a finally block would look much better